src/custom/components/swap/EthFlow/index.tsx

```diff
@@ -131,25 +118,36 @@ export function EthWethWrap({
     wrapError,
     setWrapSubmitted,
     setWrapError,
-  } = useEthFlowStatesAndSetters({ chainId, approvalState, wrapState })
+  } = useEthFlowStatesAndSetters({ chainId, approvalState })
+
+  const { currencies } = useDerivedSwapInfo()
+  const {
+    isNativeIn,
+    isNativeOut,
+    isWrappedIn,
+    isWrappedOut,
+    native,
+    wrappedToken: wrapped,
+  } = useDetectNativeToken(currencies, chainId)
+  const isNativeInSwap = isNativeIn && !isWrappedOut
+  const isNativeOutSwap = isNativeOut && !isWrappedIn
 
   const needsApproval = approvalState === ApprovalState.NOT_APPROVED
-  const needsWrap = !!wrapCallback
 
   // BALANCES
   const [nativeBalance, wrappedBalance] = useCurrencyBalances(account, [native, wrapped])
 
   // wrap/unwrap/native/wrapped related constants
   const { isWrap, isUnwrap, isNative, wrappedSymbol, nativeSymbol } = useMemo(() => {
-    const isWrap = !isNativeIn && isWrappedOut
+    const isWrap = !isNativeInSwap && isWrappedOut
```

Comment:
Won't this also be true if it's ERC20 <> WETH?
isNativeInSwap is FALSE when isNativeIn is false, so any non native token IN would return true

should be:
```suggestion
    const isWrap = isNativeIn && isWrappedOut
```

Reply:
Done: https://github.com/cowprotocol/cowswap/pull/1049/commits/c17b9decc6570e42cdea00f039ba335d9116c23c